rbac/management/utils.py

```diff
@@ -103,6 +103,7 @@ def build_system_user_from_token(request, token_validator: TokenValidator) -> Op
         system_users: dict[str, SystemUserConfig] = settings.SYSTEM_USERS
         if user and user.user_id in system_users:
             system_user = system_users[user.user_id]
+            user.username = user.user_id
```

Comment:
Should we use the preferred_username claim?

Reply:
Oh hrm I'm guessing service account tokens don't have this claim set?

Reply:
We might want to only set username if it is not already set then.

Reply:
I swear I read that! 😅 

Yeah it looks like third party org service accounts do have this claim, but not centrally managed / first party ones.

Reply:
Got it, ok! Thanks for the context. Updated with a condition for that.